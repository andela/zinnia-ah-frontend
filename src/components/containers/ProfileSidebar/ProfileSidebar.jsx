/* eslint-disable no-console */
/* eslint-disable react/prop-types */
import React from 'react';
import { Icon } from 'semantic-ui-react';

import Avatar from '../../presentationals/Avatar/Avatar.jsx';
import Button from '../../presentationals/Button/Button.jsx';
import { DEFAULT_PROFILE_PICTURE } from '../../../config/config.js';

import './ProfileSidebar.scss';

const CameraIcon = () => <Icon name="camera" />;
const OverlayImage = () => (
  <div className="overlay-image">
    <CameraIcon />
  </div>
);

export default function ProfileSidebar(props) {
  const { followers, followings, username, email } = props.profile;
  return (
    <div className="sidebar">
      <div className="avatar-div">
        <Avatar
          url={props.profile.image || DEFAULT_PROFILE_PICTURE}
          className="avatar-xl"
        />
        <OverlayImage />
      </div>
      <p className="text-center username">{username}</p>
      <p className="text-center email">{email}</p>
      <div className="follow-div">
        <div className="follow-stats">
          <p className="text-center count">
            {followings !== undefined ? followings.length : 0}
          </p>
          <p className="text-center type">Followings</p>
        </div>
        <div>
          <p className="text-center count">
            {followers !== undefined ? followers.length : 0}
          </p>
          <p className="text-center type">Followers</p>
        </div>
      </div>
      <div style={buttonStyles}>
        <Button type="submit" value="FOLLOW" className="btn-white" />
      </div>
      <hr className="sidebar-hr" />
    </div>
  );
}

const buttonStyles = {
  paddingLeft: '15%',
  paddingRight: '15%',
  marginTop: '35px',
};
