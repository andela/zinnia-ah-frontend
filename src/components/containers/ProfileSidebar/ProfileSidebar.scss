@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento');
@import '../../../assets/variables';

.sidebar {
  width: 25%;

  .avatar-div {
    display: flex;
    justify-content: center;
    margin: 20px 0;
  }

  .username {
    font-family: 'Lora', serif;
    font-size: 20px;
    font-style: normal;
    font-weight: normal;
    line-height: 26px;
    margin-bottom: 15px;
  }

  .email {
    font-family: 'Lato', sans-serif;
    font-size: 14px;
    font-style: normal;
    font-weight: normal;
    line-height: 17px;
    margin-bottom: 15px;
  }

  .sidebar-hr {
    margin-top: 30px;
    width: 75%;
  }
}

.follow-div {
  color: $secondary-black;
  display: flex;
  font-family: 'Lato', sans-serif;
  font-size: 14px;
  font-style: normal;
  font-weight: normal;
  justify-content: space-around;
  line-height: 17px;
  margin-bottom: 15px;

  .count {
    font-size: 15px;
    margin-bottom: 0;
  }
}

.text-center {
  text-align: center;
}

@media only screen and (max-width: 600px) {

  .sidebar {
    width: 100%;

    .sidebar-hr {
      display: none;
    }
  }
}
