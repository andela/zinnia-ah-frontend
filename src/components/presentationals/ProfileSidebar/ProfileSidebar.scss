@import '../../../assets/variables';

.sidebar {
  width: 25%;

  .avatar-div {
    display: flex;
    justify-content: center;
    margin: 1.25rem 0;
  }

  .username {
    font-family: $lora;
    font-size: 1.25rem;
    font-style: normal;
    font-weight: normal;
    line-height: 1.625rem;
    margin-bottom: .9375rem;
  }

  .email {
    font-family: $lato;
    font-size: .875rem;
    font-style: normal;
    font-weight: normal;
    line-height: 1.0625rem;
    margin-bottom: .9375rem;
  }

  .sidebar-hr {
    margin-top: 1.875rem;
    width: 75%;
  }
}

.follow-div {
  color: $secondary-black;
  display: flex;
  font-family: $lato;
  font-size: .875rem;
  font-style: normal;
  font-weight: normal;
  justify-content: space-around;
  line-height: 1.0625rem;
  margin-bottom: .9375rem;

  .count {
    font-size: .9375rem;
    margin-bottom: 0;
  }
}

.text-center {
  text-align: center;
}

@media only screen and (max-width: 37.5rem) {

  .sidebar {
    width: 100%;

    .sidebar-hr {
      display: none;
    }
  }
}
