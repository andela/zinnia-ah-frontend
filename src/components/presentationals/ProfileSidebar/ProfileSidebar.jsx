import React from 'react';
import PropTypes from 'prop-types';

// components
import Avatar from '../Avatar/Avatar';
import Button from '../Button/Button';

// styles
import './ProfileSidebar.scss';

// config
import { DEFAULT_USER_IMAGE_URL } from '../../../utils/config';

const ProfileSidebar = ({ followers, followings, username, email, image }) => {
  return (
    <div className="sidebar">
      <div className="avatar-div">
        <Avatar url={image || DEFAULT_USER_IMAGE_URL} className="avatar-xl" />
      </div>
      <p className="text-center username">{username}</p>
      <p className="text-center email">{email}</p>
      <div className="follow-div">
        <div className="follow-stats">
          <p className="text-center count">
            {followings !== undefined ? followings.length : 0}
          </p>
          <p className="text-center type">Followings</p>
        </div>
        <div>
          <p className="text-center count">
            {followers !== undefined ? followers.length : 0}
          </p>
          <p className="text-center type">Followers</p>
        </div>
      </div>
      <div
        style={{
          paddingLeft: '15%',
          paddingRight: '15%',
          marginTop: '35px',
        }}
      >
        <Button className="btn-white" value="FOLLOW" type="submit" />
      </div>
      <hr className="sidebar-hr" />
    </div>
  );
};

ProfileSidebar.propTypes = {
  followers: PropTypes.array,
  followings: PropTypes.array,
  username: PropTypes.string,
  email: PropTypes.string,
  image: PropTypes.string,
};

export default ProfileSidebar;
