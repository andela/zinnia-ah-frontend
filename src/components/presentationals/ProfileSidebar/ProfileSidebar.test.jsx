import React from 'react';
import { mount } from 'enzyme';

import ProfileSidebar from './ProfileSidebar';

const profile = {
  followers: [],
  followings: [],
  username: '',
  email: '',
  image: '',
};

const undefinedProfile = {
  username: '',
  email: '',
  image: '',
};

describe('ProfileSidebar', () => {
  it('should render correctly', () => {
    const component = mount(<ProfileSidebar {...profile} />);
    expect(component).toMatchSnapshot();
  });

  it('should not render correctly', () => {
    const component = mount(<ProfileSidebar {...undefinedProfile} />);
    expect(component).toMatchSnapshot();
  });
});
