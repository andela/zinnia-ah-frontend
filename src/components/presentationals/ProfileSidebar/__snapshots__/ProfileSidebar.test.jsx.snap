// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSidebar should not render correctly 1`] = `
<ProfileSidebar
  email=""
  image=""
  username=""
>
  <div
    className="sidebar"
  >
    <div
      className="avatar-div"
    >
      <Avatar
        className="avatar-xl"
        url="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
      >
        <img
          alt="avatar"
          className="avatar avatar-xl"
          src="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
        />
      </Avatar>
    </div>
    <p
      className="text-center username"
    />
    <p
      className="text-center email"
    />
    <div
      className="follow-div"
    >
      <div
        className="follow-stats"
      >
        <p
          className="text-center count"
        >
          0
        </p>
        <p
          className="text-center type"
        >
          Followings
        </p>
      </div>
      <div>
        <p
          className="text-center count"
        >
          0
        </p>
        <p
          className="text-center type"
        >
          Followers
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "35px",
          "paddingLeft": "15%",
          "paddingRight": "15%",
        }
      }
    >
      <Button
        className="btn-white"
        type="submit"
        value="FOLLOW"
      >
        <button
          className="btn btn-white"
          type="submit"
        >
          FOLLOW
        </button>
      </Button>
    </div>
    <hr
      className="sidebar-hr"
    />
  </div>
</ProfileSidebar>
`;

exports[`ProfileSidebar should render correctly 1`] = `
<ProfileSidebar
  email=""
  followers={Array []}
  followings={Array []}
  image=""
  username=""
>
  <div
    className="sidebar"
  >
    <div
      className="avatar-div"
    >
      <Avatar
        className="avatar-xl"
        url="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
      >
        <img
          alt="avatar"
          className="avatar avatar-xl"
          src="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
        />
      </Avatar>
    </div>
    <p
      className="text-center username"
    />
    <p
      className="text-center email"
    />
    <div
      className="follow-div"
    >
      <div
        className="follow-stats"
      >
        <p
          className="text-center count"
        >
          0
        </p>
        <p
          className="text-center type"
        >
          Followings
        </p>
      </div>
      <div>
        <p
          className="text-center count"
        >
          0
        </p>
        <p
          className="text-center type"
        >
          Followers
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "35px",
          "paddingLeft": "15%",
          "paddingRight": "15%",
        }
      }
    >
      <Button
        className="btn-white"
        type="submit"
        value="FOLLOW"
      >
        <button
          className="btn btn-white"
          type="submit"
        >
          FOLLOW
        </button>
      </Button>
    </div>
    <hr
      className="sidebar-hr"
    />
  </div>
</ProfileSidebar>
`;
